pandas/io/formats/info.py

```diff
@@ -1,18 +1,28 @@
 import sys
-from typing import IO, Optional, Union
+from typing import IO, TYPE_CHECKING, Optional, Tuple, Union
 
 from pandas._config import get_option
 
-from pandas._typing import FrameOrSeries
+from pandas._typing import Dtype, FrameOrSeries
 
 from pandas.io.formats import format as fmt
 from pandas.io.formats.printing import pprint_thing
 
+if TYPE_CHECKING:
+    from pandas.core.indexes.api import Index  # noqa: F401
+    from pandas.core.series import Series  # noqa: F401
 
-def _put_str(s, space):
+
+def _put_str(s: Union[str, Dtype], space: int) -> str:
     return str(s)[:space].ljust(space)
 
 
+def _get_ids_and_dtypes(data: FrameOrSeries) -> Tuple["Index", "Series"]:
```

Comment:
where is this useful?

Reply:
@jreback if I rebase #31796 onto this, the diff is easier to read as all that needs to change for this part is a
```python
if isinstance(data, Series)
```
inside this helper function

Reply:
ok, please add a doc-string